File: 프로그래머스/lv2/76502. 괄호 회전하기/괄호 회전하기.java
import java.util.*;

class Solution {
    public int solution(String s) {
        int answer = 0;
        String original = s;
        
        for(int i = 0, len = s.length(); i < len; i++){
            char[] c = s.toCharArray();
            answer += isRight(c);
            String temp = s.substring(1, len);
            temp += (s.charAt(0) +"");
            s = temp;
        }
        
        return answer;
    }
    public int isRight(char[] list) {
        Stack<Character> stack = new Stack<>();
        
        for(int i = 0, len = list.length; i < len; i++){
            if(stack.isEmpty()){
                stack.push(list[i]);
            } else{
                if(stack.peek() == '(' && list[i] == ')') {
                    stack.pop();
                }
                else if(stack.peek() == '{' && list[i] == '}') {
                    stack.pop();
                }
                else if(stack.peek() == '[' && list[i] == ']') {
                    stack.pop();
                }
                else{ 
                    stack.push(list[i]);}
            }
        }
        if(stack.isEmpty()) return 1;
        return 0;
    }
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function bracketCheck(s){
    let pair = {
        '(' : ')',
        '{' : '}',
        '[' : ']',
    }
    let stack = [];

    for(bracket of s){
        if(bracket == '(' || bracket == '{' || bracket == '['){
            stack.push(bracket)
        } else{
            if(stack.length <= 0) return false;
            if(pair[stack.pop()] !== bracket) return false;
        }
    }
    return stack.length == 0 ? true : false;
}

function solution(s) {
    let answer = 0;
    let len = s.length;
    
    for(let i = 0; i < len; i++){
        answer += bracketCheck(s) ? 1 : 0;
        s = s.at(-1) + s.slice(0, -1);
        
    }
    return answer;
}